tsconfig.json

```diff
@@ -0,0 +1,110 @@
+{
```

Comment:
1. Might be a good idea to extend from node tsconfig (i.e. [@tsconfig/node20](https://www.npmjs.com/package/@tsconfig/node20))
2. I think we should specify `outDir` and avoid checking-in the `.js` and `.d.ts` generated from typescript-compile
3. How are you commenting out anything in this file? JSON doesn't support comments afaik. Is tsconfig technically not JSON-compliant? Also commenting so many lines in one file is messy imo, but it might help the newbies discover more options.

Reply:
1. The tsconfig was generated from the older version of node. I will generate a new version of the tsconfig after the version of node has been installed. (seems like the tsconfig generated is based on the version of node, never knew that, but cool to know)

2. Typically I add the generated output directory to the .gitignore for the repo so we dont push any big, bulky compiled files into the repo when they are not needed. Typically building the production version is part of the deployment process. Not sure how we plan to handle this in our pipeline.

3. Also part of the automatically generated tsconfig file. I can remove the comments if there are any present in the new version, but it is also nice to be able to see the different settings available for those who were not involved in the dev ops aspect of this project if they wanted to learn.

Reply:
Updated tsconfig.
To add files to git ignore in future.